{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "2751e6e8-f0c4-4835-872a-0ffff459062a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from gtfparse import read_gtf\n",
    "import functools as ft\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "58780289-78c9-4617-ab38-7b98e0a955c4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'gene_type', 'gene_name', 'level', 'mgi_id', 'havana_gene', 'transcript_id', 'transcript_type', 'transcript_name', 'transcript_support_level', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'protein_id', 'ccdsid', 'ont']\n"
     ]
    }
   ],
   "source": [
    "# Read GTF with gene annotation\n",
    "genes = read_gtf('/DATA/users/magnitov/znf143/genome/gencode.vM25.annotation.gtf')\n",
    "genes = genes[genes['feature'] == 'gene']\n",
    "genes = genes[['seqname', 'start', 'end', 'strand', 'gene_id', 'gene_type', 'gene_name']]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ced7f349-2720-4cbf-b4d3-72ea5e3cd88e",
   "metadata": {},
   "source": [
    "## Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "156eee93-decb-40dc-8291-bbcf800b721d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_counts_table(genes, sample, replicate):\n",
    "    counts_fwd = pd.read_csv('/DATA/users/magnitov/znf143/ttseq/counts/counts_forward.ZFP143_TTseq.txt', \n",
    "                         names = ['gene_id', 'Zfp143_DMSO_rep1', 'Zfp143_DMSO_rep2', 'Zfp143_2H_rep1', 'Zfp143_2H_rep2', \n",
    "                                  'Zfp143_6H_rep1', 'Zfp143_6H_rep2', 'Zfp143_24H_rep1', 'Zfp143_24H_rep2'],\n",
    "                         sep = '\\t', header = None)\n",
    "    counts_fwd = counts_fwd[counts_fwd['gene_id'].isin([x for x in counts_fwd['gene_id'] if x[0]!='_'])]\n",
    "    counts_fwd = counts_fwd[['gene_id', sample + '_' + replicate]]\n",
    "    \n",
    "    counts_rev = pd.read_csv('/DATA/users/magnitov/znf143/ttseq/counts/counts_reverse.ZFP143_TTseq.txt', \n",
    "                         names = ['gene_id', 'Zfp143_DMSO_rep1', 'Zfp143_DMSO_rep2', 'Zfp143_2H_rep1', 'Zfp143_2H_rep2', \n",
    "                                  'Zfp143_6H_rep1', 'Zfp143_6H_rep2', 'Zfp143_24H_rep1', 'Zfp143_24H_rep2'],\n",
    "                         sep = '\\t', header = None)\n",
    "    counts_rev = counts_rev[counts_rev['gene_id'].isin([x for x in counts_rev['gene_id'] if x[0]!='_'])]\n",
    "    counts_rev = counts_rev[['gene_id', sample + '_' + replicate]]\n",
    "    \n",
    "    genes_merged = genes.merge(counts_fwd, on = 'gene_id', how = 'left')\n",
    "    genes_merged = genes_merged.merge(counts_rev, on = 'gene_id', how = 'left')\n",
    "    \n",
    "    counts_new = []\n",
    "    for gene in genes_merged.values:\n",
    "        if gene[3] == '+':\n",
    "            counts_new.append(gene[7])\n",
    "        else:\n",
    "            counts_new.append(gene[8])\n",
    "    genes_merged[f'{sample}_{replicate}'] = counts_new\n",
    "    \n",
    "    genes_merged = genes_merged.fillna(0)\n",
    "    \n",
    "    genes_merged = genes_merged[['seqname', 'start', 'end', 'strand', 'gene_id', 'gene_type', 'gene_name', f'{sample}_{replicate}']]\n",
    "    \n",
    "    return(genes_merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "4158e825-0815-4243-9b7f-f3956e018a7d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_type</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Zfp143_DMSO_rep1</th>\n",
       "      <th>Zfp143_DMSO_rep2</th>\n",
       "      <th>Zfp143_2H_rep1</th>\n",
       "      <th>Zfp143_2H_rep2</th>\n",
       "      <th>Zfp143_6H_rep1</th>\n",
       "      <th>Zfp143_6H_rep2</th>\n",
       "      <th>Zfp143_24H_rep1</th>\n",
       "      <th>Zfp143_24H_rep2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSMUSG00000102693.1</td>\n",
       "      <td>TEC</td>\n",
       "      <td>4933401J01Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSMUSG00000064842.1</td>\n",
       "      <td>snRNA</td>\n",
       "      <td>Gm26206</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3205901</td>\n",
       "      <td>3671498</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSMUSG00000051951.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>2</td>\n",
       "      <td>17</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3252757</td>\n",
       "      <td>3253236</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSMUSG00000102851.1</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>Gm18956</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3365731</td>\n",
       "      <td>3368549</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSMUSG00000103377.1</td>\n",
       "      <td>TEC</td>\n",
       "      <td>Gm37180</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  seqname    start      end strand               gene_id  \\\n",
       "0    chr1  3073253  3074322      +  ENSMUSG00000102693.1   \n",
       "1    chr1  3102016  3102125      +  ENSMUSG00000064842.1   \n",
       "2    chr1  3205901  3671498      -  ENSMUSG00000051951.5   \n",
       "3    chr1  3252757  3253236      +  ENSMUSG00000102851.1   \n",
       "4    chr1  3365731  3368549      -  ENSMUSG00000103377.1   \n",
       "\n",
       "              gene_type      gene_name  Zfp143_DMSO_rep1  Zfp143_DMSO_rep2  \\\n",
       "0                   TEC  4933401J01Rik                 0                 0   \n",
       "1                 snRNA        Gm26206                 0                 0   \n",
       "2        protein_coding           Xkr4                 2                17   \n",
       "3  processed_pseudogene        Gm18956                 0                 0   \n",
       "4                   TEC        Gm37180                 0                 0   \n",
       "\n",
       "   Zfp143_2H_rep1  Zfp143_2H_rep2  Zfp143_6H_rep1  Zfp143_6H_rep2  \\\n",
       "0               0               0               0               0   \n",
       "1               0               0               0               0   \n",
       "2              11              13               3              10   \n",
       "3               0               0               0               0   \n",
       "4               0               0               0               0   \n",
       "\n",
       "   Zfp143_24H_rep1  Zfp143_24H_rep2  \n",
       "0                0                0  \n",
       "1                0                0  \n",
       "2                8               16  \n",
       "3                0                0  \n",
       "4                0                0  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_Zfp143_DMSO_rep1 = get_counts_table(genes, 'Zfp143_DMSO', 'rep1')\n",
    "counts_Zfp143_DMSO_rep2 = get_counts_table(genes, 'Zfp143_DMSO', 'rep2')\n",
    "counts_Zfp143_2H_rep1 = get_counts_table(genes, 'Zfp143_2H', 'rep1')\n",
    "counts_Zfp143_2H_rep2 = get_counts_table(genes, 'Zfp143_2H', 'rep2')\n",
    "counts_Zfp143_6H_rep1 = get_counts_table(genes, 'Zfp143_6H', 'rep1')\n",
    "counts_Zfp143_6H_rep2 = get_counts_table(genes, 'Zfp143_6H', 'rep2')\n",
    "counts_Zfp143_24H_rep1 = get_counts_table(genes, 'Zfp143_24H', 'rep1')\n",
    "counts_Zfp143_24H_rep2 = get_counts_table(genes, 'Zfp143_24H', 'rep2')\n",
    "\n",
    "count_tables = [counts_Zfp143_DMSO_rep1, counts_Zfp143_DMSO_rep2, counts_Zfp143_2H_rep1, counts_Zfp143_2H_rep2, \n",
    "                counts_Zfp143_6H_rep1, counts_Zfp143_6H_rep2, counts_Zfp143_24H_rep1, counts_Zfp143_24H_rep2]\n",
    "counts_zfp143 = ft.reduce(lambda left, right: pd.merge(left, right, on = ['seqname', 'start', 'end', 'strand', 'gene_id', 'gene_type', 'gene_name']), count_tables)\n",
    "for column in counts_zfp143.columns[7:]:\n",
    "    counts_zfp143[column] = [int(x) for x in counts_zfp143[column]]\n",
    "counts_zfp143.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "434f2274-2325-47ca-8844-3649b0fb7e1b",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_zfp143.drop(['seqname', 'start', 'end', 'strand', 'gene_type', 'gene_name'], axis = 1).to_csv('/DATA/users/magnitov/znf143/ttseq/counts/counts_deseq2.txt', sep = '\\t', header = 0, index = 0)\n",
    "counts_zfp143.to_csv('/DATA/users/magnitov/znf143/ttseq/counts/counts_analysis.txt', sep = '\\t', header = 1, index = 0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "203a20e3-e23f-4c47-b429-df653850deb3",
   "metadata": {},
   "source": [
    "## Transcripts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "ba61ca52-5c63-4b2c-a1e6-5dfa15c96cec",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "INFO:root:Extracted GTF attributes: ['gene_id', 'gene_type', 'gene_name', 'level', 'mgi_id', 'havana_gene', 'transcript_id', 'transcript_type', 'transcript_name', 'transcript_support_level', 'tag', 'havana_transcript', 'exon_number', 'exon_id', 'protein_id', 'ccdsid', 'ont']\n"
     ]
    }
   ],
   "source": [
    "# Read GTF with gene annotation\n",
    "genes = read_gtf('/DATA/users/magnitov/znf143/genome/gencode.vM25.annotation.gtf')\n",
    "genes = genes[genes['feature'] == 'transcript']\n",
    "genes = genes[['seqname', 'start', 'end', 'strand', 'transcript_id', 'transcript_type', 'gene_name']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "da291974-f507-49f0-8d8b-5d1afeea325a",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_counts_table(genes, sample, replicate):\n",
    "    counts_fwd = pd.read_csv('/DATA/users/magnitov/znf143/ttseq/counts/counts_transcripts_forward.ZFP143_TTseq.txt', \n",
    "                         names = ['transcript_id', 'Zfp143_DMSO_rep1', 'Zfp143_DMSO_rep2', 'Zfp143_2H_rep1', 'Zfp143_2H_rep2', \n",
    "                                  'Zfp143_6H_rep1', 'Zfp143_6H_rep2', 'Zfp143_24H_rep1', 'Zfp143_24H_rep2'],\n",
    "                         sep = '\\t', header = None)\n",
    "    counts_fwd = counts_fwd[counts_fwd['transcript_id'].isin([x for x in counts_fwd['transcript_id'] if x[0]!='_'])]\n",
    "    counts_fwd = counts_fwd[['transcript_id', sample + '_' + replicate]]\n",
    "    \n",
    "    counts_rev = pd.read_csv('/DATA/users/magnitov/znf143/ttseq/counts/counts_transcripts_reverse.ZFP143_TTseq.txt', \n",
    "                         names = ['transcript_id', 'Zfp143_DMSO_rep1', 'Zfp143_DMSO_rep2', 'Zfp143_2H_rep1', 'Zfp143_2H_rep2', \n",
    "                                  'Zfp143_6H_rep1', 'Zfp143_6H_rep2', 'Zfp143_24H_rep1', 'Zfp143_24H_rep2'],\n",
    "                         sep = '\\t', header = None)\n",
    "    counts_rev = counts_rev[counts_rev['transcript_id'].isin([x for x in counts_rev['transcript_id'] if x[0]!='_'])]\n",
    "    counts_rev = counts_rev[['transcript_id', sample + '_' + replicate]]\n",
    "    \n",
    "    genes_merged = genes.merge(counts_fwd, on = 'transcript_id', how = 'left')\n",
    "    genes_merged = genes_merged.merge(counts_rev, on = 'transcript_id', how = 'left')\n",
    "    \n",
    "    counts_new = []\n",
    "    for gene in genes_merged.values:\n",
    "        if gene[3] == '+':\n",
    "            counts_new.append(gene[7])\n",
    "        else:\n",
    "            counts_new.append(gene[8])\n",
    "    genes_merged[f'{sample}_{replicate}'] = counts_new\n",
    "    \n",
    "    genes_merged = genes_merged.fillna(0)\n",
    "    \n",
    "    genes_merged = genes_merged[['seqname', 'start', 'end', 'strand', 'transcript_id', 'transcript_type', 'gene_name', f'{sample}_{replicate}']]\n",
    "    \n",
    "    return(genes_merged)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "66932cb6-c920-4e14-ab9c-72f5a715d57a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seqname</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_type</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>Zfp143_DMSO_rep1</th>\n",
       "      <th>Zfp143_DMSO_rep2</th>\n",
       "      <th>Zfp143_2H_rep1</th>\n",
       "      <th>Zfp143_2H_rep2</th>\n",
       "      <th>Zfp143_6H_rep1</th>\n",
       "      <th>Zfp143_6H_rep2</th>\n",
       "      <th>Zfp143_24H_rep1</th>\n",
       "      <th>Zfp143_24H_rep2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3073253</td>\n",
       "      <td>3074322</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSMUST00000193812.1</td>\n",
       "      <td>TEC</td>\n",
       "      <td>4933401J01Rik</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3102016</td>\n",
       "      <td>3102125</td>\n",
       "      <td>+</td>\n",
       "      <td>ENSMUST00000082908.1</td>\n",
       "      <td>snRNA</td>\n",
       "      <td>Gm26206</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3205901</td>\n",
       "      <td>3216344</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSMUST00000162897.1</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3206523</td>\n",
       "      <td>3215632</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSMUST00000159265.1</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3214482</td>\n",
       "      <td>3671498</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSMUST00000070533.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>2</td>\n",
       "      <td>17</td>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  seqname    start      end strand         transcript_id  \\\n",
       "0    chr1  3073253  3074322      +  ENSMUST00000193812.1   \n",
       "1    chr1  3102016  3102125      +  ENSMUST00000082908.1   \n",
       "2    chr1  3205901  3216344      -  ENSMUST00000162897.1   \n",
       "3    chr1  3206523  3215632      -  ENSMUST00000159265.1   \n",
       "4    chr1  3214482  3671498      -  ENSMUST00000070533.4   \n",
       "\n",
       "        transcript_type      gene_name  Zfp143_DMSO_rep1  Zfp143_DMSO_rep2  \\\n",
       "0                   TEC  4933401J01Rik                 0                 0   \n",
       "1                 snRNA        Gm26206                 0                 0   \n",
       "2  processed_transcript           Xkr4                 0                 0   \n",
       "3  processed_transcript           Xkr4                 0                 0   \n",
       "4        protein_coding           Xkr4                 2                17   \n",
       "\n",
       "   Zfp143_2H_rep1  Zfp143_2H_rep2  Zfp143_6H_rep1  Zfp143_6H_rep2  \\\n",
       "0               0               0               0               0   \n",
       "1               0               0               0               0   \n",
       "2               0               0               0               0   \n",
       "3               0               0               0               0   \n",
       "4              11              13               3              10   \n",
       "\n",
       "   Zfp143_24H_rep1  Zfp143_24H_rep2  \n",
       "0                0                0  \n",
       "1                0                0  \n",
       "2                0                0  \n",
       "3                0                0  \n",
       "4                8               16  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counts_Zfp143_DMSO_rep1 = get_counts_table(genes, 'Zfp143_DMSO', 'rep1')\n",
    "counts_Zfp143_DMSO_rep2 = get_counts_table(genes, 'Zfp143_DMSO', 'rep2')\n",
    "counts_Zfp143_2H_rep1 = get_counts_table(genes, 'Zfp143_2H', 'rep1')\n",
    "counts_Zfp143_2H_rep2 = get_counts_table(genes, 'Zfp143_2H', 'rep2')\n",
    "counts_Zfp143_6H_rep1 = get_counts_table(genes, 'Zfp143_6H', 'rep1')\n",
    "counts_Zfp143_6H_rep2 = get_counts_table(genes, 'Zfp143_6H', 'rep2')\n",
    "counts_Zfp143_24H_rep1 = get_counts_table(genes, 'Zfp143_24H', 'rep1')\n",
    "counts_Zfp143_24H_rep2 = get_counts_table(genes, 'Zfp143_24H', 'rep2')\n",
    "\n",
    "count_tables = [counts_Zfp143_DMSO_rep1, counts_Zfp143_DMSO_rep2, counts_Zfp143_2H_rep1, counts_Zfp143_2H_rep2, \n",
    "                counts_Zfp143_6H_rep1, counts_Zfp143_6H_rep2, counts_Zfp143_24H_rep1, counts_Zfp143_24H_rep2]\n",
    "counts_zfp143 = ft.reduce(lambda left, right: pd.merge(left, right, on = ['seqname', 'start', 'end', 'strand', 'transcript_id', 'transcript_type', 'gene_name']), count_tables)\n",
    "for column in counts_zfp143.columns[7:]:\n",
    "    counts_zfp143[column] = [int(x) for x in counts_zfp143[column]]\n",
    "counts_zfp143.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c0df9834-c9c6-4a95-b618-b4afb0c1a267",
   "metadata": {},
   "outputs": [],
   "source": [
    "counts_zfp143.drop(['seqname', 'start', 'end', 'strand', 'transcript_type', 'gene_name'], axis = 1).to_csv('/DATA/users/magnitov/znf143/ttseq/counts/counts_transcripts_deseq2.txt', sep = '\\t', header = 0, index = 0)\n",
    "counts_zfp143.to_csv('/DATA/users/magnitov/znf143/ttseq/counts/counts_transcripts_analysis.txt', sep = '\\t', header = 1, index = 0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40abcfed-b299-41f2-9556-248fcd2cf381",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:anaconda3]",
   "language": "python",
   "name": "conda-env-anaconda3-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
